### ISSUE TITLE
Bug: ... V3.0.0 exited with exit code 2

### ISSUE THREAD

**Issue by Frepke** [qdm12/private-internet-access-docker#199]
**TLDR**: *V3.0.0 exited with exit code 2*

1. Is this urgent?

    - [x] Yes
    - [ ] No

2. What VPN service provider are you using?

    - [ ] PIA
    - [ ] Mullvad
    - [ ] Windscribe
    - [x] Surfshark
    - [ ] Cyberghost

3. What's the version of the program?

    **See the line at the top of your logs**

    `Running version latest built on 2020-07-16T01:45:34Z (commit 9abb630)`

4. What are you using to run the container?

    - [ ] Docker run
    - [x] Docker Compose
    - [ ] Kubernetes
    - [ ] Docker stack
    - [ ] Docker swarm
    - [ ] Podman
    - [ ] Other:

5. Extra information

Logs:

```gluetun        | OpenVPN settings:
gluetun        | |--User: [redacted]
gluetun        | |--Password: [redacted]
gluetun        | |--Verbosity level: 1
gluetun        | |--Run as root: no
gluetun        | |--Surfshark settings:
gluetun        |  |--Network protocol: udp
gluetun        |  |--Region: netherlands amsterdam
gluetun        |  |--Target IP address: ������������Y.�h
gluetun        | System settings:
gluetun        | |--User ID: 1000
gluetun        | |--Group ID: 100
gluetun        | |--Timezone: europe/amsterdam
gluetun        | |--IP Status filepath: /ip
gluetun        | DNS over TLS disabled, using plaintext DNS 162.252.172.57
gluetun        | Firewall settings:
gluetun        |  |--Allowed subnets: 10.54.1.0/24
gluetun        | TinyProxy settings: disabled
gluetun        | ShadowSocks settings:
gluetun        |  |--Password: [redacted]
gluetun        |  |--Log: enabled
gluetun        |  |--Port: 8388
gluetun        |  |--Method: chacha20-ietf-poly1305
gluetun        | Public IP check period: 12h0m0s
gluetun        | 
gluetun        | 2020-07-16T10:07:42.130+0200   INFO    openvpn configurator: checking for device /dev/net/tun
gluetun        | 2020-07-16T10:07:42.132+0200   WARN    TUN device is not available: open /dev/net/tun: no such file or directory
gluetun        | 2020-07-16T10:07:42.132+0200   INFO    openvpn configurator: creating /dev/net/tun
gluetun        | 2020-07-16T10:07:42.132+0200   INFO    firewall: enabling...
gluetun        | 2020-07-16T10:07:42.132+0200   INFO    Launching standard output merger
gluetun        | 2020-07-16T10:07:42.133+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-07-16T10:07:42.133+0200   INFO    routing: local subnet found: 172.17.0.0/16
gluetun        | 2020-07-16T10:07:42.159+0200   INFO    firewall: enabled successfully
gluetun        | 2020-07-16T10:07:42.159+0200   INFO    firewall: setting allowed subnets through firewall...
gluetun        | 2020-07-16T10:07:42.160+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-07-16T10:07:42.160+0200   INFO    routing: local subnet found: 172.17.0.0/16
gluetun        | 2020-07-16T10:07:42.164+0200   INFO    routing: adding 10.54.1.0/24 as route via 172.17.0.1 eth0
gluetun        | 2020-07-16T10:07:42.167+0200   INFO    http server: listening on 0.0.0.0:8000
gluetun        | 2020-07-16T10:07:42.167+0200   INFO    dns over tls: falling back on plaintext DNS at address 162.252.172.57
gluetun        | 2020-07-16T10:07:42.167+0200   INFO    dns configurator: using DNS address 162.252.172.57 internally
gluetun        | 2020-07-16T10:07:42.167+0200   INFO    dns configurator: using DNS address 162.252.172.57 system wide
gluetun        | 2020-07-16T10:07:42.168+0200   INFO    firewall: setting VPN connections through firewall...
gluetun        | 2020-07-16T10:07:42.169+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-07-16T10:07:42.166+0200   INFO    shadowsocks configurator: generating configuration file
gluetun        | 2020-07-16T10:07:42.172+0200   INFO    openvpn configurator: starting openvpn
gluetun        | 2020-07-16T10:07:42.172+0200   INFO    firewall: setting allowed port 8388 through firewall...
gluetun        | 2020-07-16T10:07:42.176+0200   INFO    openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
gluetun        | 2020-07-16T10:07:42.176+0200   INFO    openvpn: library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
gluetun        | 2020-07-16T10:07:42.178+0200   INFO    shadowsocks configurator: starting shadowsocks server
gluetun        | 2020-07-16T10:07:42.182+0200   INFO    shadowsocks: UDP relay enabled
gluetun        | 2020-07-16T10:07:42.182+0200   INFO    shadowsocks: initializing ciphers... chacha20-ietf-poly1305
gluetun        | 2020-07-16T10:07:42.182+0200   INFO    shadowsocks: using nameserver: 162.252.172.57
gluetun        | 2020-07-16T10:07:42.182+0200   INFO    shadowsocks: tcp server listening at 0.0.0.0:8388
gluetun        | 2020-07-16T10:07:42.182+0200   INFO    shadowsocks: udp server listening at 0.0.0.0:8388
gluetun        | 2020-07-16T10:07:42.183+0200   INFO    openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]89.46.223.104:1194
gluetun        | 2020-07-16T10:07:42.183+0200   INFO    openvpn: UDP link local: (not bound)
gluetun        | 2020-07-16T10:07:42.183+0200   INFO    openvpn: UDP link remote: [AF_INET]89.46.223.104:1194
gluetun        | 2020-07-16T10:07:42.211+0200   WARN    openvpn: 'link-mtu' is used inconsistently, local='link-mtu 1633', remote='link-mtu 1581'
gluetun        | 2020-07-16T10:07:42.212+0200   WARN    openvpn: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher AES-256-GCM'
gluetun        | 2020-07-16T10:07:42.212+0200   WARN    openvpn: 'auth' is used inconsistently, local='auth SHA512', remote='auth [null-digest]'
gluetun        | 2020-07-16T10:07:42.212+0200   INFO    openvpn: [nl-ams-v049.prod.surfshark.com] Peer Connection Initiated with [AF_INET]89.46.223.104:1194
gluetun        | 2020-07-16T10:07:43.344+0200   ERROR   openvpn: Unrecognized option or missing or extra parameter(s) in [PUSH-OPTIONS]:7: block-outside-dns (2.4.9)
gluetun        | 2020-07-16T10:07:43.346+0200   INFO    openvpn: TUN/TAP device tun0 opened
gluetun        | 2020-07-16T10:07:43.346+0200   INFO    openvpn: /sbin/ip link set dev tun0 up mtu 1500
gluetun        | 2020-07-16T10:07:43.350+0200   INFO    openvpn: /sbin/ip addr add dev tun0 10.8.8.10/24 broadcast 10.8.8.255
gluetun        | 2020-07-16T10:07:43.360+0200   INFO    openvpn: UID set to nonrootuser
gluetun        | 2020-07-16T10:07:43.360+0200   INFO    openvpn: Initialization Sequence Completed
gluetun        | panic: runtime error: invalid memory address or nil pointer dereference
gluetun        | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7ec0f2]
gluetun        | 
gluetun        | goroutine 74 [running]:
gluetun        | github.com/qdm12/private-internet-access-docker/internal/dns.(*looper).RunRestartTicker(0xc00037a100, 0xb0cfe0, 0xc0003203c0)
gluetun        |        /tmp/gobuild/internal/dns/loop.go:271 +0xf2
gluetun        | created by main._main.func2
gluetun        |        /tmp/gobuild/main.go:173 +0xb6
gluetun exited with code 2

```

Configuration file:

```yml

```

Host OS: Debian Buster

**Comment by qdm12** [comment]
Oh well let's do v3.0.1 that was quick :smile:

**Comment by Frepke** [comment]
![afbeelding](https://user-images.githubusercontent.com/6153559/87669063-97c73900-c76d-11ea-895b-3a81c2f424ff.png)

**Comment by Frepke** [comment]
Maybe we need V3.0.2


```gluetun        | 2020-07-16T14:20:20.687+0200   INFO    openvpn configurator: checking for device /dev/net/tun
gluetun        | 2020-07-16T14:20:20.687+0200   WARN    TUN device is not available: open /dev/net/tun: no such file or directory
gluetun        | 2020-07-16T14:20:20.687+0200   INFO    openvpn configurator: creating /dev/net/tun
gluetun        | 2020-07-16T14:20:20.688+0200   INFO    firewall: enabling...
gluetun        | 2020-07-16T14:20:20.688+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-07-16T14:20:20.688+0200   INFO    routing: local subnet found: 172.17.0.0/16
gluetun        | 2020-07-16T14:20:20.688+0200   INFO    Launching standard output merger
gluetun        | 2020-07-16T14:20:20.724+0200   INFO    firewall: enabled successfully
gluetun        | 2020-07-16T14:20:20.724+0200   INFO    firewall: setting allowed subnets through firewall...
gluetun        | 2020-07-16T14:20:20.724+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-07-16T14:20:20.725+0200   INFO    routing: local subnet found: 172.17.0.0/16
gluetun        | 2020-07-16T14:20:20.729+0200   INFO    routing: adding 10.54.1.0/24 as route via 172.17.0.1 eth0
gluetun        | 2020-07-16T14:20:20.735+0200   INFO    dns over tls: falling back on plaintext DNS at address 162.252.172.57
gluetun        | 2020-07-16T14:20:20.735+0200   INFO    dns configurator: using DNS address 162.252.172.57 internally
gluetun        | 2020-07-16T14:20:20.735+0200   INFO    shadowsocks configurator: generating configuration file
gluetun        | 2020-07-16T14:20:20.735+0200   INFO    dns configurator: using DNS address 162.252.172.57 system wide
gluetun        | 2020-07-16T14:20:20.735+0200   INFO    http server: listening on 0.0.0.0:8000
gluetun        | 2020-07-16T14:20:20.736+0200   INFO    firewall: setting VPN connections through firewall...
gluetun        | 2020-07-16T14:20:20.736+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-07-16T14:20:20.740+0200   INFO    openvpn configurator: starting openvpn
gluetun        | 2020-07-16T14:20:20.740+0200   INFO    firewall: setting allowed port 8388 through firewall...
gluetun        | 2020-07-16T14:20:20.746+0200   INFO    shadowsocks configurator: starting shadowsocks server
gluetun        | 2020-07-16T14:20:20.749+0200   INFO    shadowsocks: UDP relay enabled
gluetun        | 2020-07-16T14:20:20.749+0200   INFO    shadowsocks: initializing ciphers... chacha20-ietf-poly1305
gluetun        | 2020-07-16T14:20:20.750+0200   INFO    shadowsocks: using nameserver: 162.252.172.57
gluetun        | 2020-07-16T14:20:20.750+0200   INFO    shadowsocks: tcp server listening at 0.0.0.0:8388
gluetun        | 2020-07-16T14:20:20.750+0200   INFO    shadowsocks: udp server listening at 0.0.0.0:8388
gluetun        | 2020-07-16T14:20:20.753+0200   INFO    openvpn: OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
gluetun        | 2020-07-16T14:20:20.754+0200   INFO    openvpn: library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
gluetun        | 2020-07-16T14:20:20.779+0200   INFO    openvpn: TCP/UDP: Preserving recently used remote address: [AF_INET]89.46.223.104:1194
gluetun        | 2020-07-16T14:20:20.779+0200   INFO    openvpn: UDP link local: (not bound)
gluetun        | 2020-07-16T14:20:20.779+0200   INFO    openvpn: UDP link remote: [AF_INET]89.46.223.104:1194
gluetun        | 2020-07-16T14:20:20.815+0200   INFO    openvpn: [nl-ams-v049.prod.surfshark.com] Peer Connection Initiated with [AF_INET]89.46.223.104:1194
gluetun        | 2020-07-16T14:20:27.342+0200   ERROR   openvpn: Unrecognized option or missing or extra parameter(s) in [PUSH-OPTIONS]:7: block-outside-dns (2.4.9)
gluetun        | 2020-07-16T14:20:27.344+0200   INFO    openvpn: TUN/TAP device tun0 opened
gluetun        | 2020-07-16T14:20:27.344+0200   INFO    openvpn: /sbin/ip link set dev tun0 up mtu 1500
gluetun        | 2020-07-16T14:20:27.347+0200   INFO    openvpn: /sbin/ip addr add dev tun0 10.8.8.15/24 broadcast 10.8.8.255
gluetun        | 2020-07-16T14:20:27.355+0200   INFO    openvpn: UID set to nonrootuser
gluetun        | 2020-07-16T14:20:27.355+0200   INFO    openvpn: Initialization Sequence Completed
gluetun        | 2020-07-16T14:20:27.356+0200   INFO    dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated
gluetun        | 2020-07-16T14:20:27.356+0200   INFO    routing: default route found: interface eth0, gateway 172.17.0.1
gluetun        | 2020-07-16T14:20:27.356+0200   INFO    Gateway VPN IP address: 89.46.223.104
gluetun        | 2020-07-16T14:20:27.591+0200   INFO    ip getter: Public IP address is 89.46.223.105
gluetun        | 2020-07-16T14:20:27.619+0200   INFO    dns configurator: downloading root key from https://raw.githubusercontent.com/qdm12/files/master/root.key.updated
gluetun        | 2020-07-16T14:20:27.632+0200   INFO    dns configurator: generating Unbound configuration
gluetun        | 2020-07-16T14:20:27.632+0200   INFO    dns configurator: 0 hostnames blocked overall
gluetun        | 2020-07-16T14:20:27.632+0200   INFO    dns configurator: 0 IP addresses blocked overall
gluetun        | 2020-07-16T14:20:27.632+0200   INFO    dns configurator: starting unbound
gluetun        | 2020-07-16T14:20:27.633+0200   INFO    dns configurator: using DNS address 127.0.0.1 internally
gluetun        | 2020-07-16T14:20:27.634+0200   INFO    dns configurator: using DNS address 127.0.0.1 system wide
gluetun        | 2020-07-16T14:20:27.724+0200   INFO    unbound: start of service (unbound 1.10.1).
gluetun        | 2020-07-16T14:20:27.936+0200   WARN    dns configurator: could not resolve github.com (try 1 of 10): lookup github.com on 162.252.172.57:53: server misbehaving
gluetun        | 2020-07-16T14:20:28.437+0200   WARN    dns configurator: could not resolve github.com (try 2 of 10): lookup github.com on 162.252.172.57:53: server misbehaving
gluetun        | 2020-07-16T14:20:28.942+0200   WARN    dns configurator: could not resolve github.com (try 3 of 10): lookup github.com on 162.252.172.57:53: server misbehaving
gluetun        | 2020-07-16T14:20:29.443+0200   WARN    dns configurator: could not resolve github.com (try 4 of 10): lookup github.com on 162.252.172.57:53: server misbehaving
gluetun        | 2020-07-16T14:20:29.945+0200   WARN    dns configurator: could not resolve github.com (try 5 of 10): lookup github.com on 162.252.172.57:53: server misbehaving
gluetun        | 2020-07-16T14:20:30.447+0200   WARN    dns configurator: could not resolve github.com (try 6 of 10): lookup github.com on 162.252.172.57:53: server misbehaving
gluetun        | 2020-07-16T14:20:30.950+0200   WARN    dns configurator: could not resolve github.com (try 7 of 10): lookup github.com on 162.252.172.57:53: server misbehaving
gluetun        | 2020-07-16T14:20:31.452+0200   WARN    dns configurator: could not resolve github.com (try 8 of 10): lookup github.com on 162.252.172.57:53: server misbehaving
gluetun        | 2020-07-16T14:20:31.954+0200   WARN    dns configurator: could not resolve github.com (try 9 of 10): lookup github.com on 162.252.172.57:53: server misbehaving
gluetun        | 2020-07-16T14:20:32.456+0200   WARN    dns configurator: could not resolve github.com (try 10 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:32.956+0200   INFO    dns over tls: falling back on plaintext DNS at address 162.252.172.57
gluetun        | 2020-07-16T14:20:32.957+0200   INFO    dns configurator: using DNS address 162.252.172.57 internally
gluetun        | 2020-07-16T14:20:32.957+0200   INFO    dns configurator: using DNS address 162.252.172.57 system wide
gluetun        | 2020-07-16T14:20:32.957+0200   WARN    unbound: stream error: read |0: file already closed
gluetun        | 2020-07-16T14:20:32.958+0200   WARN    dns over tls: Unbound does not seem to be working after 10 tries
gluetun        | 2020-07-16T14:20:32.958+0200   INFO    dns over tls: attempting restart in 10 seconds
gluetun        | 2020-07-16T14:20:42.958+0200   INFO    dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated
gluetun        | 2020-07-16T14:20:42.970+0200   INFO    dns configurator: downloading root key from https://raw.githubusercontent.com/qdm12/files/master/root.key.updated
gluetun        | 2020-07-16T14:20:42.982+0200   INFO    dns configurator: generating Unbound configuration
gluetun        | 2020-07-16T14:20:42.982+0200   INFO    dns configurator: 0 hostnames blocked overall
gluetun        | 2020-07-16T14:20:42.982+0200   INFO    dns configurator: 0 IP addresses blocked overall
gluetun        | 2020-07-16T14:20:42.983+0200   INFO    dns configurator: starting unbound
gluetun        | 2020-07-16T14:20:42.984+0200   INFO    dns configurator: using DNS address 127.0.0.1 internally
gluetun        | 2020-07-16T14:20:42.984+0200   INFO    dns configurator: using DNS address 127.0.0.1 system wide
gluetun        | 2020-07-16T14:20:43.074+0200   INFO    unbound: start of service (unbound 1.10.1).
gluetun        | 2020-07-16T14:20:43.286+0200   WARN    dns configurator: could not resolve github.com (try 1 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:43.788+0200   WARN    dns configurator: could not resolve github.com (try 2 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:44.289+0200   WARN    dns configurator: could not resolve github.com (try 3 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:44.791+0200   WARN    dns configurator: could not resolve github.com (try 4 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:45.293+0200   WARN    dns configurator: could not resolve github.com (try 5 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:45.795+0200   WARN    dns configurator: could not resolve github.com (try 6 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:46.296+0200   WARN    dns configurator: could not resolve github.com (try 7 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:46.798+0200   WARN    dns configurator: could not resolve github.com (try 8 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:47.299+0200   WARN    dns configurator: could not resolve github.com (try 9 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:47.801+0200   WARN    dns configurator: could not resolve github.com (try 10 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:48.302+0200   WARN    unbound: stream error: read |0: file already closed
gluetun        | 2020-07-16T14:20:48.302+0200   INFO    dns over tls: falling back on plaintext DNS at address 162.252.172.57
gluetun        | 2020-07-16T14:20:48.302+0200   INFO    dns configurator: using DNS address 162.252.172.57 internally
gluetun        | 2020-07-16T14:20:48.302+0200   INFO    dns configurator: using DNS address 162.252.172.57 system wide
gluetun        | 2020-07-16T14:20:48.303+0200   WARN    dns over tls: Unbound does not seem to be working after 10 tries
gluetun        | 2020-07-16T14:20:48.303+0200   INFO    dns over tls: attempting restart in 10 seconds
gluetun        | 2020-07-16T14:20:58.303+0200   INFO    dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated
gluetun        | 2020-07-16T14:20:58.323+0200   INFO    dns configurator: downloading root key from https://raw.githubusercontent.com/qdm12/files/master/root.key.updated
gluetun        | 2020-07-16T14:20:58.343+0200   INFO    dns configurator: generating Unbound configuration
gluetun        | 2020-07-16T14:20:58.343+0200   INFO    dns configurator: 0 hostnames blocked overall
gluetun        | 2020-07-16T14:20:58.343+0200   INFO    dns configurator: 0 IP addresses blocked overall
gluetun        | 2020-07-16T14:20:58.344+0200   INFO    dns configurator: starting unbound
gluetun        | 2020-07-16T14:20:58.348+0200   INFO    dns configurator: using DNS address 127.0.0.1 internally
gluetun        | 2020-07-16T14:20:58.349+0200   INFO    dns configurator: using DNS address 127.0.0.1 system wide
gluetun        | 2020-07-16T14:20:58.451+0200   INFO    unbound: start of service (unbound 1.10.1).
gluetun        | 2020-07-16T14:20:58.653+0200   WARN    dns configurator: could not resolve github.com (try 1 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:59.155+0200   WARN    dns configurator: could not resolve github.com (try 2 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:20:59.657+0200   WARN    dns configurator: could not resolve github.com (try 3 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:00.158+0200   WARN    dns configurator: could not resolve github.com (try 4 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:00.661+0200   WARN    dns configurator: could not resolve github.com (try 5 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:01.162+0200   WARN    dns configurator: could not resolve github.com (try 6 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:01.666+0200   WARN    dns configurator: could not resolve github.com (try 7 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:02.168+0200   WARN    dns configurator: could not resolve github.com (try 8 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:02.670+0200   WARN    dns configurator: could not resolve github.com (try 9 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:03.171+0200   WARN    dns configurator: could not resolve github.com (try 10 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:03.672+0200   INFO    dns over tls: falling back on plaintext DNS at address 162.252.172.57
gluetun        | 2020-07-16T14:21:03.673+0200   INFO    dns configurator: using DNS address 162.252.172.57 internally
gluetun        | 2020-07-16T14:21:03.673+0200   INFO    dns configurator: using DNS address 162.252.172.57 system wide
gluetun        | 2020-07-16T14:21:03.674+0200   WARN    dns over tls: Unbound does not seem to be working after 10 tries
gluetun        | 2020-07-16T14:21:03.674+0200   INFO    dns over tls: attempting restart in 10 seconds
gluetun        | 2020-07-16T14:21:03.672+0200   WARN    unbound: stream error: read |0: file already closed
gluetun        | 2020-07-16T14:21:13.675+0200   INFO    dns configurator: downloading root hints from https://raw.githubusercontent.com/qdm12/files/master/named.root.updated
gluetun        | 2020-07-16T14:21:13.694+0200   INFO    dns configurator: downloading root key from https://raw.githubusercontent.com/qdm12/files/master/root.key.updated
gluetun        | 2020-07-16T14:21:13.717+0200   INFO    dns configurator: generating Unbound configuration
gluetun        | 2020-07-16T14:21:13.718+0200   INFO    dns configurator: 0 hostnames blocked overall
gluetun        | 2020-07-16T14:21:13.718+0200   INFO    dns configurator: 0 IP addresses blocked overall
gluetun        | 2020-07-16T14:21:13.720+0200   INFO    dns configurator: starting unbound
gluetun        | 2020-07-16T14:21:13.722+0200   INFO    dns configurator: using DNS address 127.0.0.1 internally
gluetun        | 2020-07-16T14:21:13.722+0200   INFO    dns configurator: using DNS address 127.0.0.1 system wide
gluetun        | 2020-07-16T14:21:13.845+0200   INFO    unbound: start of service (unbound 1.10.1).
gluetun        | 2020-07-16T14:21:14.027+0200   WARN    dns configurator: could not resolve github.com (try 1 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:14.532+0200   WARN    dns configurator: could not resolve github.com (try 2 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:15.036+0200   WARN    dns configurator: could not resolve github.com (try 3 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:15.537+0200   WARN    dns configurator: could not resolve github.com (try 4 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:16.043+0200   WARN    dns configurator: could not resolve github.com (try 5 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:16.546+0200   WARN    dns configurator: could not resolve github.com (try 6 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:17.048+0200   WARN    dns configurator: could not resolve github.com (try 7 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:17.551+0200   WARN    dns configurator: could not resolve github.com (try 8 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:18.054+0200   WARN    dns configurator: could not resolve github.com (try 9 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:18.556+0200   WARN    dns configurator: could not resolve github.com (try 10 of 10): lookup github.com on 127.0.0.1:53: server misbehaving
gluetun        | 2020-07-16T14:21:19.056+0200   INFO    dns over tls: falling back on plaintext DNS at address 162.252.172.57
gluetun        | 2020-07-16T14:21:19.056+0200   INFO    dns configurator: using DNS address 162.252.172.57 internally
gluetun        | 2020-07-16T14:21:19.057+0200   INFO    dns configurator: using DNS address 162.252.172.57 system wide
gluetun        | 2020-07-16T14:21:19.057+0200   WARN    dns over tls: Unbound does not seem to be working after 10 tries
gluetun        | 2020-07-16T14:21:19.057+0200   INFO    dns over tls: attempting restart in 10 seconds
gluetun        | 2020-07-16T14:21:19.056+0200   WARN    unbound: stream error: read |0: file already closed

**Comment by qdm12** [comment]
Ah yes they plaintext dns needs fixing. I'll do that this evening, sorry for the interruption. Feel free to rebuild from a few days ago until then.

**Comment by Frepke** [comment]
No worries 😃

**Comment by qdm12** [comment]
I tested it quickly with `-e DOT=off` and `-e DNS_UPDATE_PERIOD=0` it looks like it's working. Let me know if it works, and I'll make that shiny v3.0.2 😄

**Comment by Frepke** [comment]
In the latest build ```2020-07-17T01:17:21Z (commit fab9939)``` it works both,
```-e DOT=off and -e DNS_UPDATE_PERIOD=0``` and 
```-e DOT=off and -e DNS_UPDATE_PERIOD=24h```
only 1 error in the log for now ```gluetun        | 2020-07-17T06:33:25.314+0200	ERROR	openvpn: Unrecognized option or missing or extra parameter(s) in [PUSH-OPTIONS]:7: block-outside-dns (2.4.9)```

**Comment by qdm12** [comment]
Cool! I think that's fine, see https://askubuntu.com/a/987538 I'll probably add it to the logging blacklist as it's irrelevant. Although is it crashing openvpn (I guess not but maybe)?

**Comment by Frepke** [comment]
OpenVPN is running for 6 hours now without problems.

**Comment by qdm12** [comment]
Made the tag `v3.1.0`, sorry I forgot about v3.0.2 for a few days 😄
